src/Search/SearchWithSuggestions/SearchWithSuggestions.js

```diff
@@ -171,37 +260,187 @@ class SearchWithSuggestions extends PureComponent {
+      <>
+        {!noData && (
+          <ViewAllResults
+            searchTerm={searchTerm}
+            onViewAllResults={onViewAllResults}
+            suggestions={suggestions}
+          />
+        )}
+        {!noData &&
+          types.map((key, index) => {
+            const { label, options } = suggestions[key]
+
+            const formattedOptions = getSliced(options)
+
+            if (!formattedOptions.length) {
+              return null
+            }
+
+            fromCounter += formattedOptions.length
+
+            return (
+              <Fragment key={key}>
+                {label && <div className={styles.groupLabel}>{label}</div>}
+                <SuggestionItemsList
+                  fromCounter={fromCounter - formattedOptions.length}
+                  suggestions={formattedOptions}
+                  cursor={cursor}
+                  onSuggestionSelect={selected =>
+                    onSuggestionSelect([key, selected])
+                  }
+                  suggestionContent={suggestionContent}
+                />
+                {index !== types.length - 1 && (
+                  <div className={styles.divider} />
+                )}
+              </Fragment>
+            )
+          })}
+        {noData && <NoResults isSearching={isSearching} />}
+      </>
+    )
+  } else {
+    const sliced = getSliced(suggestions)
+    return sliced.length > 0 ? (
+      <>
+        <ViewAllResults
+          searchTerm={searchTerm}
+          onViewAllResults={onViewAllResults}
+          suggestions={suggestions}
+        />
+        <SuggestionItemsList
+          suggestions={sliced}
+          cursor={cursor}
+          onSuggestionSelect={onSuggestionSelect}
+          suggestionContent={suggestionContent}
+        />
+      </>
+    ) : (
+      <NoResults isSearching={isSearching} />
+    )
+  }
+}
+
+const ViewAllResults = ({ searchTerm, onViewAllResults, suggestions }) => {
+  if (!searchTerm || !onViewAllResults) {
+    return null
+  }
+
+  return (
+    <>
+      <div
+        className={styles.viewAllResults}
+        onMouseDown={() => {
+          onViewAllResults(searchTerm, suggestions)
+        }}
+      >
+        View all results for “{searchTerm}”
+      </div>
+      <div className={styles.divider} />
+    </>
+  )
+}
+
+const NoResults = ({ isSearching }) => (
+  <div className={styles.suggestion + ' ' + styles.noresults}>
+    {!isSearching ? 'No results found' : 'Searching...'}
+  </div>
+)
+
+const SuggestionItemsList = ({
+  suggestions,
+  cursor,
+  onSuggestionSelect,
+  suggestionContent,
+  fromCounter = 0
+}) => {
+  return suggestions.map((suggestion, index) => {
+    const isActive = index + fromCounter === cursor
+
+    return (
+      <SuggestionItem
+        key={index}
+        isActive={isActive}
+        onSuggestionSelect={onSuggestionSelect}
+        suggestionContent={suggestionContent}
+        suggestion={suggestion}
+      />
+    )
+  })
+}
+
+const SuggestionItem = ({
```

Comment:
move `SuggestionItem`, `SuggestionItemsList` and other additional components to separate files